production/dpl-workflow.sh

```diff
@@ -231,8 +241,14 @@ elif [ $EXTINPUT == 1 ]; then
   PROXY_INSPEC="dd:FLP/DISTSUBTIMEFRAME/0;eos:***/INFORMATION"
   PROXY_IN_N=0
   for i in `echo "$WORKFLOW_DETECTORS" | sed "s/,/ /g"`; do
-    if [ $EPNMODE == 1 ] && [ $i == "TOF" ]; then
-      PROXY_INTYPE=CRAWDATA
+    if has_detector_flp_processing $i; then
+      case $i in
+        TOF)
+          PROXY_INTYPE=CRAWDATA;;
+        *)
+          echo Input type for detector $i with FLP processing not defined 1>&2
+          exit 1;;
+      esac
```

Comment:
this loop with 1 input per detector will not work for all detectors except TOF, perhaps we should add a function which can be called from multiple times in the single loop, something like

```
add_to_proxy_input()
{
  if (( $# < 2 )); then
    echo "destination and at least 1 input to add is expected"
    exit 1
  fi
  for ((i = 2; i <= $#; i++ )); do
    count=`grep -o ':' <<< "${!1}"  | wc -l
    $1+=";INP${count}:{i}"
  }
}
```
and use it as 

```
  for i in `echo "$WORKFLOW_DETECTORS" | sed "s/,/ /g"`; do
    if has_detector_flp_processing $i; then
      case $i in
        TOF)
          add_to_proxy_input PROXY_INSPEC "$i/CRAWDATA" ;;
        FT0 | FV0 | FDD )
          add_to_proxy_input PROXY_INSPEC "$i/DIGITSBC"  "$i/DIGITSCH" ;;
        PHS)
          add_to_proxy_input PROXY_INSPEC "$i/CELLS"  "$i/CELLTRIGREC" ;;
        CPV)
          add_to_proxy_input PROXY_INSPEC "$i/DIGITS"  "$i/DIGITTRIGREC" "$i/RAWHWERRORS" ;;
        EMC)
          add_to_proxy_input PROXY_INSPEC "$i/CELLS"  "$i/CELLSTRGR" "$i/DECODERERR" ;;
        *)
          echo Input type for detector $i with FLP processing not defined 1>&2
          exit 1;;
      esac
    else
      add_to_proxy_input PROXY_INSPEC "$i/RAWDATA"
    fi
  done
```

For the QC, we have 2 additions:
```
[ -z "$QC_JSON_FT0" ] && QC_JSON_FT0=/home/afurs/work/epn/configs/qc/ft0-digits-qc-ds.json
[ -z "$QC_JSON_FV0" ] && QC_JSON_FV0=/home/afurs/work/epn/configs/qc/fv0-digits-qc-ds.json
```